### DISCUSSION TITLE
How can I create a TTree with nested branches?

### DISCUSSION THREAD

**Discussion by jmcarcell** [scikit-hep/uproot5#1132] [Q&A] [answered]
I want to create a TTree with a certain structure so a class can be read later with ROOT.
From the tutorial we can do the following to create a TTree:
```
file["tree1"] = {"branch1": np.arange(1000), "branch2": np.arange(1000)*1.1}
```
What if I wanted branch1 to belong to one top-level branch and branch2 to belong to a different top-level branch instead of how currently they will be both a top level branch inside `tree1`?

**Comment by jpivarski** [comment, marked as answer]
I didn't add a way to do that because I didn't see what it would be useful for. Nesting TBranch objects doesn't have anything to do with how nested the data are.

Also, it would be difficult to implement. Since the WritableTree object needs to be mutable, to `extend` it iteratively over a large dataset, it's implemented as a template with values that can change (like the array of pointers to TBaskets). That template is general enough to include a flat list of TBranches, but not a nested list. (It's the _metadata_ that gets nested, but the metadata was implemented as a template, and it would be hard to generalize the template.)
